#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_gcc4.3_missing_includes.dpatch by Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add missing includes to fix FTBFS with gcc 4.3

@DPATCH@

--- a/src/yafraycore/HDR_io.cc	2007-12-21 02:04:16.000000000 +0100
+++ b/src/yafraycore/HDR_io.cc	2007-12-21 02:04:23.000000000 +0100
@@ -9,6 +9,8 @@
 #include<config.h>
 #endif
 
+#include <cstring>
+
 __BEGIN_YAFRAY
 
 //---------------------------------------------------------------------------
--- a/src/yafraycore/buffer.h	2007-12-21 02:03:39.000000000 +0100
+++ b/src/yafraycore/buffer.h	2007-12-21 02:03:48.000000000 +0100
@@ -29,6 +29,7 @@
 
 #include <cstdio>
 #include <iostream>
+#include <cstdlib>
 #include "color.h"
 
 __BEGIN_YAFRAY
--- a/src/yafraycore/ccthreads.cc	2007-12-21 02:05:02.000000000 +0100
+++ b/src/yafraycore/ccthreads.cc	2007-12-21 02:05:12.000000000 +0100
@@ -1,5 +1,6 @@
 #include"ccthreads.h"
 #include<iostream>
+#include<cstdlib>
 
 using namespace std;
 
--- a/src/yafraycore/geometree.h	2007-12-21 02:09:40.000000000 +0100
+++ b/src/yafraycore/geometree.h	2007-12-21 02:09:46.000000000 +0100
@@ -2,6 +2,7 @@
 #define __GEOMETREE_H
 
 #include"bound.h"
+#include <limits>
 
 __BEGIN_YAFRAY
 
--- a/src/yafraycore/kdtree.cc	2007-12-21 02:05:55.000000000 +0100
+++ b/src/yafraycore/kdtree.cc	2007-12-21 02:06:03.000000000 +0100
@@ -4,6 +4,7 @@
 #include <math.h>
 #include <limits>
 #include <time.h>
+#include <cstring>
 
 __BEGIN_YAFRAY
 
--- a/src/yafraycore/renderblock.cc	2007-12-21 02:07:41.000000000 +0100
+++ b/src/yafraycore/renderblock.cc	2007-12-21 02:07:49.000000000 +0100
@@ -1,5 +1,6 @@
 
 #include "renderblock.h"
+#include <cstdlib>
 
 using namespace std;
 
--- a/src/loader/mlex.cc	2007-12-21 02:13:25.000000000 +0100
+++ b/src/loader/mlex.cc	2007-12-21 02:13:35.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include<stdio.h>
 #include<iostream>
+#include<cstring>
 #include "mlex.h"
 using namespace std;
 #include<string>
--- a/src/shaders/basictex.cc	2007-12-21 02:14:43.000000000 +0100
+++ b/src/shaders/basictex.cc	2007-12-21 02:14:52.000000000 +0100
@@ -23,6 +23,7 @@
 #include "basictex.h"
 #include "object3d.h"
 #include <iostream>
+#include <cstring>
 
 #include "targaIO.h"
 #include "HDR_io.h"
--- a/src/yafraycore/targaIO.cc	2007-12-21 02:10:53.000000000 +0100
+++ b/src/yafraycore/targaIO.cc	2007-12-21 02:11:00.000000000 +0100
@@ -3,6 +3,7 @@
 //--------------------------------------------------------------------------------
 #include "targaIO.h"
 #include "vector3d.h"
+#include <cstring>
 
 //--------------------------------------------------------------------------------
 // Save uncompressed 24 bit targa
